part-3/part-3.md

```diff
@@ -152,252 +153,289 @@ const source = from([1, 2, 3]).pipe(
+sourceTakeFirst.subscribe(value => console.log(value));
 ```
 
-Отлично! Теперь мы умеем брать несколько первых или несколько последних значений из потока. Давайте посмотрим, как еще можно фильтровать значения с помощью операторов **RxJS**.
+Great!Now we know how to take the first values from a stream. Let's see how else can we filter values using **RxJS** operators.
+
+## **distinct()**
+
+Sometimes we need to only handle unique values, or in other words, ignore duplicates. For that purposes we can utilize the **distinct** operator. It will allow only those values that have not been emitted yet.
+
+```ts
+import { from } from 'rxjs';
+
+import { distinct } from 'rxjs/operators';
+
+const source = from([1, 2, 2, 1, 3, 2, 3, 1, 3, 2, 1, 1, 3]).pipe(
+  // Will skip duplicates
+
+  distinct(),
+);
+
+// Will log:
+
+//. 1
+
+//. 2
+
+//. 3
+
+source.subscribe(value => console.log(value));
+```
 
 ## **distinctUntilChanged()**
 
-Иногда, нам может потребоваться фильтровать значения из потока, которые повторяются друг за другом. Для этого можно использовать оператор **distinctUntilChanged()**.
+Sometimes though, we will need to skip duplicates, but only if they come one after another. For that, we can use the  **distinctUntilChanged()** operator.
 
-**distinctUntilChanged()** убирает повторяющиеся друг за другом значения в потоке.
+**distinctUntilChanged()** skips duplicate values that immediately follow each other.
 
 ```ts
-import { from } from "rxjs";
+import { from } from 'rxjs';
 
-import { distinctUntilChanged } from "rxjs/operators";
+import { distinctUntilChanged } from 'rxjs/operators';
 
 const source = from([1, 1, 2, 2, 2, 3, 3, 1, 1]).pipe(
-  // Не пропустит значение, если оно было предыдущим
+  // Will skip the value, if it is a duplicate of the previous one
 
-  distinctUntilChanged()
+  distinctUntilChanged(),
 );
 
-// Выведет в консоль:
+// Will log:
 
 //. 1
 
 //. 2
 
-// 3
+//. 3
 
 //. 1
 
-source.subscribe((value) => console.log(value));
+source.subscribe(value => console.log(value));
 ```
 
-Как видно из примера, **distinctUntilChanged()** пропустил только первые из повторяющихся значений.
+As we can see, **distinctUntilChanged()** skipped only the first duplicates.
 
 #
 
-# **Объединение потоков**
+# **Combining streams**
 
-Иногда бывает так, что нам необходимо правильно обработать значения сразу с нескольких потоков. Для начала давайте просто их создадим, как это делали раньше. Для этого воспользуемся оператором **timer()**.
+Sometimes we deal with tasks that require us to handle several streams simultaneously. For starters, let's create some streams to work with. For that purpose we will use the  **timer()** function.
 
 ## **timer()**
 
-**timer()** создает поток, который будет возвращать новые значения с таймаутом.
+**timer()** creates a stream that emits a value the time (in milliseconds) we provide has passed, essentially working like `setTimeout`, but for streams.
```

Comment:
```suggestion
**timer()** creates a stream that emits a value every time the interval (in milliseconds) we provide has passed, essentially working like `setInterval`, but for streams.
```

Reply:
`timer` is like `setTimeout` followed by `setInterval` when it's passed 2 arguments since it adds an initial delay as specified by the first argument. `interval` is like `setInterval`.

How do we express this?

Reply:
I don't know, it wa written in a sloppy way in the original, so I just removed it :D

Reply:
I'll leave a thread about this in [our Part 3 discussion](https://github.com/this-is-learning/rxjs-fundamentals-course/discussions/3).